### PR TITLE
Issue-1398: set umask when launching browsers

### PR DESCRIPTION
**Pull request by therealdjryan**
**Thanks for contributing to the Docker-Selenium project!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines, applied for this repository.
Avoid large PRs, help reviewers by making them as simple and short as possible.

Set umask to 002 before launching browsers. This will cause downloaded files to 664 (-rw-rw-r)
https://github.com/SeleniumHQ/docker-selenium/issues/1398

### Motivation and Context

Allows changing downloaded files when run in an environment where the userids don't map

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://selenium.dev/documentation/en/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by diemol — COMMENTED** [review]
Would this fix #1095?

**Review by diemol — COMMENTED** [review]
Also, I know we do not have a wrapper for Firefox, but are files downloaded with Firefox also affected by #1398?

File: NodeChrome/wrap_chrome_binary [hunk 1]
```diff
@@ -6,6 +6,7 @@ mv "$WRAPPER_PATH" "$BASE_PATH"
 
 cat > "$WRAPPER_PATH" <<_EOF
 #!/bin/bash
+umask 002
```
Comment (diemol) [inline NodeChrome/wrap_chrome_binary:11]: Can we leave a comment explaining the reasoning of the command placed here?

File: NodeEdge/wrap_edge_binary [hunk 2]
```diff
@@ -6,7 +6,7 @@ mv "$WRAPPER_PATH" "$BASE_PATH"
 
 cat > "$WRAPPER_PATH" <<_EOF
 #!/bin/bash
-
+umask 002
```
Comment (diemol) [inline NodeEdge/wrap_edge_binary:11]: Can we leave a comment explaining the reasoning of the command placed here?

**Review by diemol — CHANGES_REQUESTED** [review]

**Comment by gjoc96** [comment]
James, all the process has been done and reporting will be fetched as mentioned.
Attached SIT file will be soon uploaded. 
Merging can be performed once request change are addressed.

**Comment by therealdjryan** [comment]
> Would this fix #1095?

This is one of the symptoms I saw which caused me to look for a solution. I'm not actually very happy with the solution I've come up with as it is too high level, I would prefer 002 to be the default for all user processes, or at least for seluser's. I did try to modify /etc/profile and seluser's profile, but I couldn't get it to work. Maybe the real solution is to make the change to supervisord.conf, but I don't have much experience with supervisord (I was a sys admin when Solaris was king!)

**Comment by therealdjryan** [comment]
> Also, I know we do not have a wrapper for Firefox, but are files downloaded with Firefox also affected by #1398?

I believe it firefox will also have this problem, but we don't use Firefox in our environment

File: NodeEdge/wrap_edge_binary (hunk 2 above)
Comment (therealdjryan) [reply NodeEdge/wrap_edge_binary:11]: @diemol, I have to confess that I have not tested this change with Edge as it is not used in our environment.

**Comment by therealdjryan** [comment]
@diemol, I believe I have addressed the changes you requested.

**Comment by diemol** [comment]
Apologies for the delay reviewing this PR, we've been busy with the 4.0 release. We will review this in the next couple of days. Thanks!

**Comment by therealdjryan** [comment]
No problem and congratulations! I've been running 4 fairly successfully for
a few weeks now.

**Comment by diemol** [comment]
@therealdjryan I was going to start reviewing this but seems something went wrong with the latest commits. Why are there so many changes in the this PR?

**Comment by therealdjryan** [comment]
@diemol wow that's ugly! Pretty sure all I did was to sync the fork in IntelliJ. I'll ditch this fork and create a clean one.

**Comment by therealdjryan** [comment]
@diemol, I updated my fork using the github UI and my commits look normal again.

**Review by diemol — APPROVED** [review]
This does not fix #1095 because it is a different issue. 

Thanks for contributing, @therealdjryan!

**Pull request merged**